src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -242,6 +242,7 @@ public void start(Map<String, String> properties) {
       }
     }
 
+    disableAutoCommit(cachedConnectionProvider.getConnection());
```

Comment:
`CachedConnectionProvider#getConnection())` performs a validity-test on the connection instance it caches before returning it.  This may result in a dead connection being discarded and replaced with a new instance:

https://github.com/confluentinc/kafka-connect-jdbc/blob/66307690724f6fe54d0a2bb28bbfe8b30fc6a447/src/main/java/io/confluent/connect/jdbc/util/CachedConnectionProvider.java#L63-L66

Due to the `start()`/`poll()` lifecycle, is it possible that a new connection without auto-commit disabled might become active?  Calls might need to be moved, or the option pushed-down into the connection-provider framework.

Reply:
Thank you for the call-out, this was a serious issue with that implementation. I've added a callback that disables autocommit for all new connections from that cached connection provider.